client/disputes/evidence/index.js

```diff
@@ -353,252 +47,85 @@ const getDisputeProductType = ( dispute ) => {
-	const [ evidence, setEvidence ] = useState( {} ); // Evidence to update.
 	const {
-		createSuccessNotice,
-		createErrorNotice,
-		createInfoNotice,
-	} = useDispatch( 'core/notices' );
-
-	const pristine =
-		! dispute ||
-		isMatchWith(
-			dispute.evidence,
-			evidence,
-			( disputeValue, formValue ) => {
-				// Treat null and '' as equal values.
-				if ( null === disputeValue && ! formValue ) {
-					return true;
-				}
-			}
-		);
-
-	const confirmationNavigationCallback = useConfirmNavigation( () => {
-		if ( pristine ) {
-			return;
-		}
-
-		return __(
-			'There are unsaved changes on this page. Are you sure you want to leave and discard the unsaved changes?',
-			'woocommerce-payments'
-		);
-	} );
-
-	useEffect( confirmationNavigationCallback, [
-		pristine,
-		confirmationNavigationCallback,
-	] );
+		isSavingEvidence,
+		isUploadingEvidence,
+		evidenceTransient,
+		evidenceUploadErrors,
+		saveEvidence,
+		submitEvidence,
+		updateEvidenceTransientForDispute,
+		updateEvidenceUploadErrorsForDispute,
+		uploadFileEvidenceForDispute,
+	} = useDisputeEvidence( disputeId );
+
+	const { setMessage: setNavigationMessage } = useConfirmNavigation();
 
 	useEffect( () => {
-		const fetchDispute = async () => {
-			setLoading( true );
-			try {
-				setDispute( await apiFetch( { path } ) );
-			} finally {
-				setLoading( false );
-			}
-		};
-
-		fetchDispute();
-	}, [ setLoading, setDispute, path ] );
-
-	const updateEvidence = ( key, value ) =>
-		setEvidence( ( e ) => ( { ...e, [ key ]: value } ) );
-	const updateDispute = ( updates = {} ) =>
-		setDispute( ( d ) => merge( {}, d, updates ) );
-	const isUploadingEvidence = () => some( dispute.isUploading );
-
-	const doRemoveFile = ( key ) => {
-		updateEvidence( key, '' );
-		updateDispute( {
-			metadata: { [ key ]: '' },
-			uploadingErrors: { [ key ]: '' },
-			fileSize: { [ key ]: 0 },
-		} );
-	};
+		const isPristine =
+			! dispute ||
+			! evidenceTransient || // Empty evidence transient means no local updates.
+			isMatchWith(
+				dispute.evidence,
+				evidenceTransient,
+				( disputeValue, formValue ) => {
+					// Treat null and '' as equal values.
+					if ( null === disputeValue && ! formValue ) {
+						return true;
+					}
+				}
+			);
 
-	const fileSizeExceeded = ( latestFileSize ) => {
-		const fileSizeLimitInBytes = 4500000;
-		const fileSizes = dispute.fileSize
-			? Object.values( dispute.fileSize )
-			: [];
-		const totalFileSize =
-			fileSizes.reduce( ( acc, fileSize ) => acc + fileSize, 0 ) +
-			latestFileSize;
-		if ( fileSizeLimitInBytes < totalFileSize ) {
-			createInfoNotice(
+		if ( isPristine ) {
+			setNavigationMessage( '' );
+		} else if ( isSavingEvidence ) {
+			// We don't want to show the confirmation message while saving evidence.
+			setNavigationMessage( '' );
+		} else if ( some( dispute.isUploading ) ) {
+			// We don't want to show the confirmation message while submitting evidence.
+			setNavigationMessage( '' );
+		} else {
```

Comment:
I added these 2 conditions since we were seeing the confirmation of wanting to discard unsaved evidence information when saving or submitting information.